Prototypes.lua

```diff
@@ -2560,15 +2560,18 @@ WeakAuras.event_prototypes = {
     },
     name = L["Chat Message"],
     init = function(trigger)
-      return [[
+      local ret = [[
         if (event:find('LEADER')) then
           event = event:sub(0, -8);
         end
         if (event == 'CHAT_MSG_TEXT_EMOTE') then
           event = 'CHAT_MSG_EMOTE';
         end
+         local use_cloneId = %s;
       ]];
+      return ret:format(trigger.use_cloneId and "true" or "false");
     end,
+    statesParameter = "all",
```

Comment:
This patch shows how simple it is for triggers to use the new trigger states.

By setting statesParameter to all, and having a cloneId the chat trigger can have clones.
And by setting "store = true" on the message and sourceName, there are additional text replacements for #message# and #sourceName#.